contracts/infrastructure/ens/ArgentENSManager.sol

```diff
@@ -87,34 +89,50 @@ contract ArgentENSManager is IENSManager, Owned, Managed {
     * Registers both the forward and reverse ENS.
     * @param _label The subdomain label.
     * @param _owner The owner of the subdomain.
+    * @param _managerSignature The manager signature of the hash of _owner and _label.
     */
-    function register(string calldata _label, address _owner) external onlyManager {
+    function register(string calldata _label, address _owner, bytes calldata _managerSignature) external override validateENSLabel(_label) {
+        bytes32 signedHash = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", keccak256(abi.encodePacked(_owner, _label))));
+        validateManagerSignature(signedHash, _managerSignature);
+
         bytes32 labelNode = keccak256(abi.encodePacked(_label));
         bytes32 node = keccak256(abi.encodePacked(rootNode, labelNode));
         address currentOwner = ensRegistry.owner(node);
-        require(currentOwner == address(0), "AEM: _label is alrealdy owned");
+        require(currentOwner == address(0), "AEM: label is already owned");
 
         // Forward ENS
         ensRegistry.setSubnodeRecord(rootNode, labelNode, _owner, address(ensResolver), 0);
         ensResolver.setAddr(node, _owner);
 
         // Reverse ENS
```

Comment:
Just an idea but maybe here we could check if the current instance is the owner of the reverse node and only set the reverse ENS if it is. This will give the users the flexibility to only register the forward ENS if they want to.

Reply:
That could be a problem for our own name resolution though as the clients are currently using the reverse resolver to check wallet address registrations. Or are you saying that functionality will just not work for them and that's okay?

Reply:
Yes, the clients shouldn't assume the reverse ENS is set for an address. And we've had users not wanting their reverse ENS to be set for privacy reasons, so I believe it would be nice to have that flexibility (on top of being cheaper).